spark/src/test/scala/org/apache/spark/sql/delta/managedcommit/ManagedCommitSuite.scala

```diff
@@ -116,6 +123,405 @@ class ManagedCommitSuite
     }
   }
 
+  test("cold snapshot initialization") {
+    val builder = TrackingInMemoryCommitStoreBuilder(batchSize = 10)
+    val commitStore = builder.build(Map.empty).asInstanceOf[TrackingCommitStore]
+    CommitStoreProvider.registerBuilder(builder)
+    withTempDir { tempDir =>
+      val tablePath = tempDir.getAbsolutePath
+      Seq(1).toDF.write.format("delta").mode("overwrite").save(tablePath) // version 0
+      DeltaLog.clearCache()
+      checkAnswer(sql(s"SELECT * FROM delta.`$tablePath`"), Seq(Row(1)))
+
+      Seq(2).toDF.write.format("delta").mode("overwrite").save(tablePath) // version 1
+      Seq(3).toDF.write.format("delta").mode("append").save(tablePath) // version 2
+      DeltaLog.clearCache()
+      commitStore.numGetCommitsCalled = 0
+      import testImplicits._
+      val result1 = sql(s"SELECT * FROM delta.`$tablePath`").collect()
+      assert(result1.length === 2 && result1.toSet === Set(Row(2), Row(3)))
+      assert(commitStore.numGetCommitsCalled === 2)
+    }
+  }
+
+  test("cold snapshot initialization - with and without initial commit store") {
+    CommitStoreProvider.registerBuilder(TrackingInMemoryCommitStoreBuilder(batchSize = 10))
+    withTempDir { tempDir =>
+      val tablePath = tempDir.getAbsolutePath
+      Seq(1).toDF.write.format("delta").mode("overwrite").save(tablePath) // version 0
+      Seq(2).toDF.write.format("delta").mode("overwrite").save(tablePath) // version 1
+      Seq(2).toDF.write.format("delta").mode("overwrite").save(tablePath) // version 2
```

Comment:
Are these repeated overwrites somehow important, out of curiosity?

Reply:
In this test case, they are not relevant. Removing these.